sycl/test-e2e/Graph/nodes_profiling_info.cpp

```diff
@@ -0,0 +1,331 @@
+    auto NodeCopyStart = CopyEvent.ext_oneapi_get_profiling_info<
+        sycl::info::event_profiling::command_start>(NodeCopy);
+    auto NodeCopyEnd = CopyEvent.ext_oneapi_get_profiling_info<
+        sycl::info::event_profiling::command_end>(NodeCopy);
+
+    assert((CopySubmit == NodeCopySubmit) && "Copy submit times differ");
+    assert((CopyStart == NodeCopyStart) && "Copy start times differ");
+    assert((CopyEnd == NodeCopyEnd) && "Copy end times differ");
+
+    // Check first execution
+
+    auto Submit =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_submit>();
+    auto Start =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_start>();
+    auto End =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_end>();
+
+    std::vector<uint64_t> NodeSubmit(Nodes.size());
+    std::vector<uint64_t> NodeStart(Nodes.size());
+    std::vector<uint64_t> NodeEnd(Nodes.size());
+    uint64_t GlobalRuntime;
+    for (int i = 0; i < Nodes.size(); i++) {
+
+      assert(verifyProfiling(KernelEvent1, Nodes[i]));
+
+      NodeSubmit[i] = KernelEvent1.ext_oneapi_get_profiling_info<
+          sycl::info::event_profiling::command_submit>(Nodes[i]);
+      NodeStart[i] = KernelEvent1.ext_oneapi_get_profiling_info<
+          sycl::info::event_profiling::command_start>(Nodes[i]);
+      NodeEnd[i] = KernelEvent1.ext_oneapi_get_profiling_info<
+          sycl::info::event_profiling::command_end>(Nodes[i]);
+
+      GlobalRuntime += (NodeEnd[i] - NodeStart[i]);
+    }
+
+    // submit time should be all the same
+    assert((Submit == NodeSubmit[0]) && "Submit times differ");
+    assert((Submit == NodeSubmit[1]) && "Submit times differ");
+    assert((Submit == NodeSubmit[2]) && "Submit times differ");
+    assert((Submit == NodeSubmit[3]) && "Submit times differ");
+
+    // check start timestamps order
+    assert((Start == NodeStart[0]) &&
+           "Graph timestamp and First node timestamp differ");
+    assert(NodeStart[0] < NodeStart[1]);
+    assert(NodeEnd[0] <= NodeStart[1]);
+    assert(NodeStart[0] < NodeStart[2]);
+    assert(NodeEnd[0] <= NodeStart[2]);
+    assert(NodeStart[1] < NodeStart[3]);
+    assert(NodeEnd[1] <= NodeStart[3]);
+    assert(NodeStart[2] < NodeStart[3]);
+    assert(NodeEnd[2] <= NodeStart[3]);
+    assert((End == NodeEnd[3]) &&
+           "Graph timestamp and last node timestamp differ");
+
+#if GRAPH_TESTS_VERBOSE_PRINT
+    std::cout << "Global Node Runtime = " << GlobalRuntime << std::endl;
+    std::cout << "Global Node Runtime timestamp based = "
+              << (NodeEnd[3] - NodeStart[0]) << std::endl;
+    std::cout << "Graph runtime = " << (End - Start) << std::endl;
+#endif
+
+    assert(((NodeEnd[3] - NodeStart[0]) == (End - Start)) &&
+           "Global Runtime differ");
+    // The graph node addtional time to handle sychornization. So the sum of
```

Comment:
```suggestion
    // The graph node additional time to handle synchronization. So the sum of
```

Reply:
I've updated this comment to make it clearer.